resources/js/components/Elements/Slider.vue

```diff
@@ -1,71 +1,115 @@
 <script>
-    export default {
-        render() {
-            return this.$scopedSlots.default({
-                navigate: this.navigate,
-                showLeft: this.showLeft,
-                showRight: this.showRight,
-                currentSlide: this.currentSlide,
-                slidesTotal: this.slidesTotal
-            })
+export default {
+    render() {
+        return this.$scopedSlots.default({
+            navigate: this.navigate,
+            showLeft: this.showLeft,
+            showRight: this.showRight,
+            currentSlide: this.currentSlide,
+            slidesTotal: this.slidesTotal,
+        })
+    },
+    props: {
+        reference: {
+            type: String,
+            default: 'slider',
         },
-        props: {
-            reference: {
-                type: String,
-                default: 'slider'
-            },
-            vertical: {
-                type: Boolean,
-                default: false
-            }
+        vertical: {
+            type: Boolean,
+            default: false,
         },
-        data: () => {
-            return {
-                position: 0,
-                showLeft: false,
-                showRight: false,
-                mounted: false
-            }
+        interval: {
+            type: Number,
+            default: false,
+        },
+    },
+    data: () => {
+        return {
+            position: 0,
+            showLeft: false,
+            showRight: false,
+            mounted: false,
+            hover: false,
+        }
+    },
+    mounted() {
+        this.slider.addEventListener('scroll', this.scroll)
+        this.slider.dispatchEvent(new CustomEvent('scroll'))
+        this.mounted = true
+        if (this.interval > 0) {
+            window.setInterval(this.autoScroll, this.interval * 1000)
+        }
+    },
+    beforeDestroy() {
+        this.slider.removeEventListener('scroll', this.scroll)
+    },
+    destroyed() {
+        if (this.interval > 0) {
+            window.clearInterval(this.autoScroll)
+        }
+    },
```

Comment:
Are you sure this is the right place? As `destroyed()` is called after the component is destroyed so `this.autoScroll` isn't available anymore right?